javatests/arcs/core/storage/ReferenceModeStoreTest.kt

```diff
@@ -254,6 +254,34 @@ class ReferenceModeStoreTest {
             .isEqualTo(bobEntity.data.collections.mapValues { it.value.data })
     }
 
+    @Test
+    fun removeOpClearsBackingEntity() = runBlockingTest {
+        DriverFactory.register(MockDriverProvider())
+
+        val activeStore = createReferenceModeStore()
+        val actor = activeStore.crdtKey
+        val bob = createPersonEntity("an-id", "bob", 42)
+
+        // Add Bob to collection.
+        val addOp = RefModeStoreOp.SetAdd(actor, VersionMap(actor to 1), bob)
+        assertThat(
+            activeStore.onProxyMessage(ProxyMessage.Operations(listOf(addOp), id = 1))
+        ).isTrue()
+        // Bob was added to the backing store.
+        val storedBob = activeStore.backingStore.getLocalData("an-id") as CrdtEntity.Data
+        assertThat(storedBob.toRawEntity("an-id")).isEqualTo(bob)
+
+        // Remove Bob from the collection.
+        val deleteOp = RefModeStoreOp.SetRemove(actor, VersionMap(actor to 1), bob)
+        assertThat(
+            activeStore.onProxyMessage(ProxyMessage.Operations(listOf(deleteOp), id = 1))
+        ).isTrue()
+
+        // Check the backing store Bob has been cleared.
+        val storedBob2 = activeStore.backingStore.getLocalData("an-id") as CrdtEntity.Data
+        assertThat(storedBob2.toRawEntity("an-id")).isEqualTo(createEmptyPersonEntity("an-id"))
+    }
+
```

Comment:
This looks great, can you add a similar test to the variants of the ReferenceModeStoreIntegrationTest?

Reply:
Yes you were right, added tests with the database impl and had to fix an issue with null values. I am not sure there are equivalent tests with ramdisks?